descartes_planner/src/dense_planner.cpp

```diff
@@ -31,10 +31,10 @@ DensePlanner::~DensePlanner()
 
 }
 
-bool DensePlanner::initialize(descartes_core::RobotModelConstPtr &model)
+bool DensePlanner::initialize(descartes_core::RobotModelConstPtr model)
 {
   planning_graph_ = boost::shared_ptr<descartes_planner::PlanningGraph>(
-      new descartes_planner::PlanningGraph(model));
+      new descartes_planner::PlanningGraph(std::move(model)));
```

Comment:
The shared pointer is "anchored" in main. That pointer is copied into the initialize method and then moved down to planning graph. When initialize is done the reference count on the pointer should be 2.